app/realm/models/client_upgrade.js

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import Realm from 'realm';
+
+export default class ClientUpgrade extends Realm.Object {
```

Comment:
Why do we need this?

Reply:
Are we not saving the last time we checked for an upgrade or is it OK in this case to keep the date in redux?

Reply:
Well I was asking to get an explanation

Reply:
Ahhh, I see!
Originally we were storing this global state for lastUpdateCheck in the reducer. To store within Realm we create a schema using this model so that Realm when opening and accessing/writing to the DB understands how to interact with the data. This is 'how' the data for this will be stored now. The name 'ClientUpgrade' defines the table and how we access it later (realm.object). The Primary ID in this case is written wrong I see now, this should be an ID and to update the latest check I just write to that ID. This will be updated in my next commit.

Reply:
Ok I see where you coming from. The theory here is correct but I'm wondering why do we need to create a whole new table just to store **one** value, it seems too much for something that in short is only available for custom builds.

I guess an alternative to this would be using the async storage, but the better question would be *Why do we even need to persist this value?*

Reply:
Thats a good question. It doesn't really matter when we did our last check, considering it's always checking to see if we need to upgrade from the looks of it.